File: Programmers/[77485] 행렬 테두리 회전하기/Johoseong/77485.py
def rotate(matrix, x1, y1, x2, y2):
    min_num = 10 ** 10
    left_tmp = matrix[x2][y1]
    right_tmp = matrix[x1][y2]
    for c in range(y2 - 1, y1 - 1, -1): # 위
        matrix[x1][c + 1] = matrix[x1][c]
        min_num = min(min_num, matrix[x1][c])
    for c in range(y1 + 1, y2 + 1): # 아래
        matrix[x2][c - 1] = matrix[x2][c]
        min_num = min(min_num, matrix[x2][c])

    for r in range(x1 + 1, x2 + 1): # 왼쪽
        matrix[r - 1][y1] = matrix[r][y1]
        min_num = min(min_num, matrix[r][y1])
    for r in range(x2 - 1, x1 - 1, -1): # 오른쪽
        matrix[r + 1][y2] = matrix[r][y2]
        min_num = min(min_num, matrix[r][y2])

    matrix[x1 + 1][y2] = right_tmp
    matrix[x2 - 1][y1] = left_tmp
    min_num = min(min_num, right_tmp)
    min_num = min(min_num, left_tmp)
    return min_num

def solution(rows, columns, queries):
    answer = []
    num = 1
    matrix = []
    for r in range(rows):
        matrix.append([])
        for _ in range(columns):
            matrix[r].append(num)
            num += 1

    for x1, y1, x2, y2 in queries:
        min_num = rotate(matrix, x1 - 1, y1 - 1, x2 - 1, y2 - 1)
        answer.append(min_num)
    return answer

File: Programmers/[77485] 행렬 테두리 회전하기/SUbbb/77485.java
class Solution {
    
    static int[][] map;
    
    public int[] solution(int rows, int columns, int[][] queries) {
        int[] answer = new int[queries.length];
        int idx = 0;
        
        init(rows, columns);
        
        for (int[] query : queries)
            answer[idx++] = rotation(query[0] - 1, query[1] - 1, query[2] - 1, query[3] - 1);
        
        return answer;
    }
    
    private void init(int rows, int columns) {
        map = new int[rows][columns];
        int num = 1;
        
        for (int i = 0; i < rows; i++)
            for (int j = 0; j < columns; j++)
                map[i][j] = num++;
    }
    
    private int rotation(int x1, int y1, int x2, int y2) {
        int w = y2 - y1;
        int h = x2 - x1;
        int value = map[x1][y1];
        int min = value;
        
        // 왼쪽 줄 이동
        for (int i = x1 + 1; i <= x1 + h; i++) {
            int moveN = map[i][y1];
            if (min > moveN) min = moveN;
            map[i - 1][y1] = moveN;
        }
        
        // 아랫 줄 이동
        for (int j = y1 + 1; j <= y1 + w; j++) {
            int moveN = map[x2][j];
            if (min > moveN) min = moveN;
            map[x2][j - 1] = moveN;
        }
        
        // 오른쪽 줄 이동
        for (int i = x2 - 1; i >= x1; i--) {
            int moveN = map[i][y2];
            if (min > moveN) min = moveN;
            map[i + 1][y2] = moveN;
        }
        
        // 윗 줄 이동
        for (int j = y2 - 1; j > y1; j--) {
            int moveN = map[x1][j];
            if (min > moveN) min = moveN;
            map[x1][j + 1] = moveN;
        }
        
        map[x1][y1 + 1] = value;
        
        return min;
    }
}